src/java.base/share/classes/java/net/URLConnection.java

```diff
@@ -1401,35 +1390,30 @@ private ContentHandler lookupContentHandlerClassFor(String contentType) {
 
     @SuppressWarnings("removal")
     private ContentHandler lookupContentHandlerViaProvider(String contentType) {
-        return AccessController.doPrivileged(
-                new PrivilegedAction<>() {
-                    @Override
-                    public ContentHandler run() {
-                        ClassLoader cl = ClassLoader.getSystemClassLoader();
-                        ServiceLoader<ContentHandlerFactory> sl =
-                                ServiceLoader.load(ContentHandlerFactory.class, cl);
-
-                        Iterator<ContentHandlerFactory> iterator = sl.iterator();
-
-                        ContentHandler handler = null;
-                        while (iterator.hasNext()) {
-                            ContentHandlerFactory f;
-                            try {
-                                f = iterator.next();
-                            } catch (ServiceConfigurationError e) {
-                                if (e.getCause() instanceof SecurityException) {
-                                    continue;
-                                }
-                                throw e;
-                            }
-                            handler = f.createContentHandler(contentType);
-                            if (handler != null) {
-                                break;
-                            }
-                        }
-                        return handler;
-                    }
-                });
+
+        ClassLoader cl = ClassLoader.getSystemClassLoader();
+        ServiceLoader<ContentHandlerFactory> sl =
+                ServiceLoader.load(ContentHandlerFactory.class, cl);
+
+        Iterator<ContentHandlerFactory> iterator = sl.iterator();
+
+        ContentHandler handler = null;
+        while (iterator.hasNext()) {
+            ContentHandlerFactory f;
+            try {
+                f = iterator.next();
+            } catch (ServiceConfigurationError e) {
+                if (e.getCause() instanceof SecurityException) {
```

Comment:
I assume the try-catch can be removed as SecurityException won't be thrown.

Reply:
Is that sure? We might have custom handlers there (ContentHandlerFactory is a public exported service interface) - some of them might be throwing `SecurityException` for reasons of their own. That's why I was conservatively keeping this code. Please let me know if you think I should just throw SCE now in that case.

Reply:
A thrown SecurityException (from a provider) would be handled as any other RuntimeException.
It is not specified that Providers can throw SecurityException and expect it to be ignored.
It may be worth a release note, though.

Reply:
I did remove a catch `SecurityException` around a call to `InetAddress.getByName` in `java.net.URL` though. We know that `InetAddress.getByName` no longer throws `SecurityException`.